[doc] Workflow for writing a Python test for #1462 (#1475)
taichi-dev/taichi: docs/utilities.rst

Logging
-------

Taichi uses `spdlog <https://github.com/gabime/spdlog>`_ as its logging system.
Logs can have different levels, from low to high, they are:

.. code-block:: none

    trace
    debug
    info
    warn
    error

The higher the level is, the more critical the message is.

The default logging level is ``info``. You may override the default logging level by:

1. Setting the environment variable like ``export TI_LOG_LEVEL=warn``.
2. Set log level from Python side: ``ti.set_logging_level(ti.WARN)``.

In Python, you may write logs using the ``ti.*`` interface:

.. code-block:: python

    # Python
    ti.trace("Hello world!")
    ti.debug("Hello world!")
    ti.info("Hello world!")
    ti.warn("Hello world!")
    ti.error("Hello world!")

In C++, you may write logs using the ``TI_*`` interface:

.. code-block:: cpp

    // C++
    TI_TRACE("Hello world!");
    TI_DEBUG("Hello world!");
    TI_INFO("Hello world!");
    TI_WARN("Hello world!");
    TI_ERROR("Hello world!");

> If one raised a message of level is ``error``, then Taichi will be **terminated** immediately

Reviewer: ```suggestion
If one raises a message of the level ``error``, Taichi will be **terminated** immediately
```
